project_app/pages/createreview.tsx

```diff
@@ -0,0 +1,139 @@
+import { useState, useContext, useEffect, useRef } from "react";
+import React from 'react';
+import { CFormFloating } from "@coreui/react";
+import "@coreui/coreui/dist/css/coreui.min.css";
+import Swal from "sweetalert2";
+import { useForm, Controller } from "react-hook-form";
+
+import AdminLayout from '../components/AdminLayout';
+
+import {StyledCFormTextarea, StyledCFormInput, StyledCFormLabel, FormLayout, FormTitle, FormDialog, DialogTitle, FormContainer, DialogText, ActionButtonContainer, ActionButton, StyledCFormSelect, StyledFormLabel} from '../components/formstyled';
+
+
+
+export default function CreateReview() {
+    const { register, handleSubmit, reset, formState: {isDirty, isValid, errors}} = useForm();
+    const [games, setGames] = useState([]); 
+    const selectedGameName = useRef(null);
+    
+    async function fetchGames() {
+        const response = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/api/games`);
+        const games = await response.json();
+        setGames(games);
+
+        // Name of game on load
+        selectedGameName.current = games[0].name;

```

Comment:
lets not use ref, should be state

Reply:
but would using state trigger re-renders if the state changes?

Reply:
yes its ok.